src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -223,25 +209,29 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
         final String stepName = step.getName();
         ProLogger.info("Simulating phase: " + stepName);
         if (stepName.endsWith("NonCombatMove")) {
+          ProData.initialize(this, dataCopy, playerCopy);
           final Map<Territory, ProTerritory> factoryMoveMap =
-              nonCombatMoveAI.doNonCombatMove(null, null, moveDel, dataCopy, playerCopy, true);
+              nonCombatMoveAI.doNonCombatMove(null, null, moveDel, true);
```

Comment:
A second method signature that would allow this to be `doNonCombatMove(moveDel)` would be of huge benefit. First code smell is the null value, second code smell is passing null as  a method arg, next code smell is the boolean parameter, and yet another with the too many arg method with 4 method parameters, 3 of which are not even used.